src/cryptsetup/cryptsetup-generator.c

```diff
@@ -49,19 +49,19 @@ static int generate_keydev_mount(const char *name, const char *keydev, char **un
         assert(mount);
 
         r = mkdir_parents("/run/systemd/cryptsetup", 0755);
-        if (r < 0)
+        if (r < 0 && r != -EEXIST)
                 return r;
 
         r = mkdir("/run/systemd/cryptsetup", 0700);
-        if (r < 0)
+        if (r < 0 && errno != EEXIST)
                 return r;
 
         where = strjoin("/run/systemd/cryptsetup/keydev-", name);
         if (!where)
                 return -ENOMEM;
 
         r = mkdir(where, 0700);
-        if (r < 0)
+        if (r < 0 && errno != EEXIST)
                 return r;
```

Comment:
Here too.

Reply:
Stupid copy paste mistakes. ```mkdir()``` and friends look similar but actually are not.